CRM/Membershipextrasimporterapi/EntityImporter/RecurContribution.php

```diff
@@ -108,8 +108,21 @@ private function getAmount() {
   }
 
   private function getCurrency() {
-    // todo : not in mapping document, need to discuss with others how to get it.
-    return 'GBP';
+    if (!isset($this->cachedValues['currencies_enabled'])) {
+      $sqlQuery = "SELECT cov.name as name, cov.value as id FROM civicrm_option_value cov 
```

Comment:
The logic for getting currency here (line 111 to 119) is  duplicated with CRM/Membershipextrasimporterapi/EntityImporter/Contribution.php line 152 and 160, should we decouple it?

Reply:
there is more duplicate code that this actually in some of the classes, I might do an overlook at all these in one go later and find a better place to put them.

Reply:
Noted